File: Q1.java
import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public class Q1{

    public int[] twoSum(int[] nums, int target){

      // Data structures needed for solution
      HashMap<Integer, Integer> map = new HashMap<>();
      int[] solution = new int[2];

      //Code to figure out what the solution is.
      for(int i = 0; i < nums.length; i++){
        if(map.containsKey(nums[i])){
          solution[0] = map.get(nums[i]);
          solution[1] = i;
        }else{
          map.put(target-nums[i],i);
        }
      }

      return solution;
    }
}

/*
Runtime: 2 ms, faster than 98.83% of Java online submissions for Two Sum.
Memory Usage: 37.8 MB, less than 98.29% of Java online submissions for Two Sum.
*/

File: Q1/Q1.py
class Solution:
	def twoSum(self, nums, target):

		# Variables for tracking
		answer = []
		dict= {}

		# Checking for membership in my dictionary and returning a list.
		for i, num in enumerate(nums):
			if num in dict:
				answer.append(dict[num])
				answer.append(i)
				break
			else:
				dict[target-num] = i

		return answer

"""
Runtime: 60 ms, faster than 68.91% of Python3 online submissions for Two Sum.
Memory Usage: 15.2 MB, less than 5.11% of Python3 online submissions for Two Sum.

"""
